src/workers.jl

```diff
@@ -105,6 +106,25 @@ function watch_and_terminate!(w::Worker, ev::Threads.Event)
     true
 end
 
+# Send signal to the given `Worker` process to trigger a profile.
+# Users can customise this profiling in the usual way, e.g. via
+# `JULIA_PROFILE_PEEK_HEAP_SNAPSHOT`, but `Profile.set_peek_duration`, `Profile.peek_report[]`
+# would have to be modified in the worker process.
+# See https://docs.julialang.org/en/v1/stdlib/Profile/#Triggered-During-Execution
+# Called when timeout_profile_wait is non-zero.
+function trigger_profile(w::Worker, timeout_profile_wait, from::Symbol=:manual)
+    if !Sys.iswindows()
+        @debug "sending profile request to worker $(w.pid) from $from"
+        if Sys.islinux()
+            kill(w.process, 10)  # SIGUSR1
+        elseif Sys.isbsd()
+            kill(w.process, 29)  # SIGINFO
+        end
+    end
+    sleep(timeout_profile_wait) # Leave time for it to print the profile.
```

Comment:
Why `sleep` on non-`islinux` or `isbsd` platforms? This can move into the `if !` block?

Reply:
Good point!